src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -513,6 +515,32 @@ class RawSourceData {
     return snapshotSources;
   }
 
+  /**
+   * @param permissions RAS dbGaP permissions held by the caller (derived from the caller's linked
+   *     RAS passports)
+   * @return snapshot UUIDs accessible under the permissions
+   */
+  @Transactional(
+      propagation = Propagation.REQUIRED,
+      isolation = Isolation.SERIALIZABLE,
+      readOnly = true)
+  public List<UUID> getAccessibleSnapshots(List<RasDbgapPermissions> permissions) {
+    String sql =
+        "SELECT snapshot.id FROM snapshot "
+            + "JOIN snapshot_source ON snapshot.id = snapshot_source.snapshot_id "
+            + "JOIN dataset ON dataset.id = snapshot_source.dataset_id "
+            + "WHERE snapshot.consent_code IS NOT NULL AND dataset.phs_id IS NOT NULL "
+            + "AND (snapshot.consent_code, dataset.phs_id) IN (:permissions)";
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue(
+                "permissions",
+                permissions.stream()
+                    .map(c -> new String[] {c.consent_group(), c.phs_id()})
```

Comment:
Does this have to be an array, or would a collection work here too? An array is fine but I think the collection version is a little easier to read, if it works.
```suggestion
                    .map(c -> List.of(c.consent_group(), c.phs_id()))
```

Reply:
The Spring documentation specifically mentioned an object array:

https://docs.spring.io/spring-framework/docs/3.0.x/spring-framework-reference/html/jdbc.html

```
12.7.3 Passing in lists of values for IN clause
…
In addition to the primitive values in the value list, you can create a java.util.List of object arrays. This list would support multiple expressions defined for the in clause such as

select * from T_ACTOR where (id, last_name) in ((1, 'Johnson'), (2, 'Harrop'))
```

But I can test out with a list, I think that should work too and agree on the readability.

Reply:
Oh, I think you're right. The current docs are a bit nicer formatted but the information seems the same. https://docs.spring.io/spring-framework/docs/current/reference/html/data-access.html#jdbc-in-clause

Reply:
Closing the loop, I tested this to be sure and confirmed that a list of arrays is required.